File: publish_examples.m
function [] = publish_examples(varargin)
% Create html files for matbiips examples, and matlab zip files
% Last update: 18/02/2017

outdir = 'docs';
if nargin>=1
    outdir = varargin{1};
end

ind_folders = 1:4;
if nargin>=2
    ind_folders = varargin{2};
end

options = struct();
if nargin>=3
    options = varargin{3};
end

folders = {...
    'tutorial',...
    'object_tracking',...
    'stoch_kinetic',...
    'stoch_volatility'...
    };
mfiles = {...
    {'tutorial1', 'tutorial2', 'tutorial3'},...
    {'hmm_4d_nonlin'},...
    {'stoch_kinetic_gill', 'stoch_kinetic'},...
    {'switch_stoch_volatility', 'stoch_volatility', 'switch_stoch_volatility_param'}...
    };

fprintf('========================================================\n')

d = dir(outdir);
outdir = d(1).folder; % absolute path
curdir = pwd;
for i = ind_folders
    mdir = fullfile(curdir, folders{i});
    cd(mdir);
    options.outputDir = fullfile(outdir, 'matbiips', folders{i});
    files_i = mfiles{i};
    for j = 1:length(files_i)
        % Publish html file
        fprintf('Publishing matbiips example: %s\n', files_i{j})
        publish([files_i{j} '.m'], options);
        close all
        fprintf('------------------------------------------------\n')         
    end
end

File: publish_examples.r
library(rmarkdown)

root_dir = getwd()

output_dir = file.path(root_dir, "docs/rbiips")

clear_cache = TRUE

example_scripts = list()
example_scripts$tutorial = c("tutorial1", "tutorial2", "tutorial3")
example_scripts$object_tracking = c("hmm_4d_nonlin")
example_scripts$stoch_kinetic = c("stoch_kinetic", "stoch_kinetic_gill")
example_scripts$stoch_volatility = c("stoch_volatility",
                                     "switch_stoch_volatility",
                                     "switch_stoch_volatility_param")

# loop over all example directories
for (ex_dir in names(example_scripts)) {
  # set example output directory
  out_dir = file.path(output_dir, ex_dir)
  dir.create(out_dir, recursive = TRUE, showWarnings=FALSE)
  # loop over all scripts in example directory
  for (ex_script in example_scripts[[ex_dir]]) {
    # change directory
    setwd(file.path(root_dir, ex_dir))
    # clear cache
    if (clear_cache)
      unlink(paste0(ex_script, "_cache"), recursive=TRUE)
    # render html
    input = paste0(ex_script, ".r")
    rmarkdown::render(input,
                      output_dir = out_dir,
                      output_format="html_document")
  }
}
